app/scripts/lib/multichain-api/provider-authorize.js

```diff
@@ -0,0 +1,130 @@
+import { EthereumRpcError } from 'eth-rpc-errors';
+import { RestrictedMethods } from '../../../../shared/constants/permissions';
+import { processScopes, mergeScopes } from './scope';
+import {
+  Caip25CaveatType,
+  Caip25EndowmentPermissionName,
+} from './caip25permissions';
+
+const getAccountsFromPermission = (permission) => {
+  return permission.eth_accounts.caveats.find(
+    (caveat) => caveat.type === 'restrictReturnedAccounts',
+  )?.value;
+};
+
+// TODO:
+// Unless the dapp is known and trusted, give generic error messages for
+// - the user denies consent for exposing accounts that match the requested and approved chains,
+// - the user denies consent for requested methods,
+// - the user denies all requested or any required scope objects,
+// - the wallet cannot support all requested or any required scope objects,
+// - the requested chains are not supported by the wallet, or
+// - the requested methods are not supported by the wallet
+// return
+//     "code": 0,
+//     "message": "Unknown error"
+
+// TODO:
+// When user disapproves accepting calls with the request methods
+//   code = 5001
+//   message = "User disapproved requested methods"
+// When user disapproves accepting calls with the request notifications
+//   code = 5002
+//   message = "User disapproved requested notifications"
+
+export async function providerAuthorizeHandler(req, res, _next, end, hooks) {
+  // TODO: Does this handler need a rate limiter/lock like the one in eth_requestAccounts?
```

Comment:
no it has it by default with the approval controller unless explicitly specified as not needing rate limiting